kmodes/util/dissim.py

```diff
@@ -7,14 +7,38 @@
 
 import numpy as np
 
-
-def matching_dissim(a, b):
+def matching_dissim(a, b, **kwargs):
     """Simple matching dissimilarity function"""
     return np.sum(a != b, axis=1)
 
 
-def euclidean_dissim(a, b):
+def euclidean_dissim(a, b, **kwargs):
     """Euclidean distance dissimilarity function"""
     if np.isnan(a).any() or np.isnan(b).any():
         raise ValueError("Missing values detected in numerical columns.")
     return np.sum((a - b) ** 2, axis=1)
+
+
+# Implementation of Ng et al.'s dissimilarity measure
+# 	Michael K. Ng, Mark Junjie Li, Joshua Zhexue Huang, and Zengyou He, "On the
+# 	Impact of Dissimilarity Measure in k-Modes Clustering Algorithm", IEEE
+# 	Transactions on Pattern Analysis and Machine Intelligence, Vol. 29, No. 3,
+# 	January, 2007
+# Author: 'Ben Andow' <beandow@ncsu.edu>
+# Note that membship must be a rectangular array such that the len(membship) = len(a) and len(membship[i]) = X.shape[1]
+def ng_dissim(a, b, X, membship):
+    """Ng et al.'s dissimilarity measure"""
+    if len(membship) != len(a) and len(membship[0]) != X.shape[1]:
+        raise ValueError("'membship' must be a rectangular array where the number of rows in 'membship' equals the number of rows in 'a' and the number of columns in 'membship' equals the number of rows in 'X'.")
+
+    return np.array([ np.array([calc_dissim(b, X, membship[idj], idr, idj) if b[idr] == t else 1.0 for idr,t in enumerate(val_a) ]).sum(0) for idj,val_a in enumerate(a) ])
+
+# Num objects w/ category value x_{i,r} for rth attr in jth cluster
+def calcCJR(b, X, memj, idr):
+    xcids = np.where(np.in1d(memj.ravel(), [1]).reshape(memj.shape))
+    return float((np.take(X, xcids, axis=0)[0][:, idr] == b[idr]).sum(0))
+
+def calc_dissim(b, X, memj, idr, idj):
```

Comment:
Good catch! I removed that parameter. I had implemented support for a weighting function, which was removed for the PR and that parameter was a leftover remnant.